interface1.js

```diff
@@ -798,6 +804,155 @@ window.addEventListener('load', function(){
 		return htmlChunks.join('');
 	}
 
+	// GENERATE TERMINAL STRINGS
+
+	// add list of terminals button
+	document.getElementById('addList').addEventListener('click', function(){
+		var length = spotForm.genStringsInput.length;
+		if(length === undefined) {
+			length = 1;
+		}
+		var newLength = length + 1;
+		length = length.toString();
+		newLength = newLength.toString();
+		document.getElementById('list'+length).insertAdjacentHTML('afterend', "<div id='list"+newLength+"'>List of terminals "+newLength+": <input type='text' name='genStringsInput'><p>Number of terminals in generated strings:</p><p class='genStringsNum'>Min: <input type='text' name='genStringsMin' class='genStringsNumBox' style='margin-left: 4px'></p><p class='genStringsNum'>Max: <input type='text' name='genStringsMax' class='genStringsNumBox'></p></div>");
+		document.getElementById('autoDoneMessage').style.display = 'none';
+	});
+
+	// show/hide generated terminal strings
+	document.getElementById('gen-strings-switch').addEventListener('click', function(){
+		if (document.getElementById('genStringsArea').style.display === 'none' && document.getElementById('gen-strings-switch').checked){
+			document.getElementById('genStringsArea').style.display = 'block';
+			document.getElementById('strings-switch-text').innerHTML = 'Hide generated terminals strings';
+		}
+		else{
+			document.getElementById('genStringsArea').style.display = 'none';
+			document.getElementById('strings-switch-text').innerHTML = 'Show generated terminals strings';
+		}
+	});
+
+	// done button for generate terminal strings
+	document.getElementById('genStringsDoneButton').addEventListener('click', function(){
+		deleteThickLine();
+		genTerminalStrings();
+		document.getElementById('genStringsArea').style.display = 'block';
+		document.getElementById('gen-strings-switch').checked = true;
+		document.getElementById('strings-switch-text').innerHTML = 'Hide generated terminals strings';
+	});
+
+	var genStringsList;
+
+	// generate and display terminal strings
+	function genTerminalStrings() {
```

Comment:
This is minor but I'm just noticing that we have genTerminalStrings() (a UI function) plus generateTerminalStrings() (a purely backend function that genTerminalStrings() calls). This is a recipe for confusion. I think it would be good to rename genTerminalStrings() to something like genTerminalStringsUI() or genTerminalStringsInterface().